plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java

```diff
@@ -244,14 +247,24 @@ public void put(final InputStream inputStream) throws IOException, NugetPackageE
       recordMetadata.put(CREATED, creationTime);
       recordMetadata.put(LAST_UPDATED, creationTime);
       recordMetadata.put(PUBLISHED, creationTime);
-
+      OrientVertex component = null;
       try (InputStream in = tempStream.get()) {
-        createOrUpdatePackage(storageTx, recordMetadata, in);
+        component = createOrUpdatePackage(storageTx, recordMetadata, in);
       }
 
-      maintainAggregateInfo(storageTx, recordMetadata.get(ID));
+      String id = recordMetadata.get(ID);
+      String version = recordMetadata.get(VERSION);
+      maintainAggregateInfo(storageTx, id);
 
+      boolean isNew = component.getIdentity().isNew();  // must check before commit
       storageTx.commit();
+
+      if (isNew) {
+        getEventBus().post(new ComponentCreatedEvent(component, getRepository()));
+      }
+      else {
+        getEventBus().post(new ComponentUpdatedEvent(component, getRepository()));
+      }
     }
```

Comment:
It's not a good idea to post external events while inside a transaction as this can lead to nested transactions, as I had to fix previously in: https://github.com/sonatype/nexus-pro/commit/e1826f10d7f62b3800cc671d01703efcd11b924a

Wrt. partial data (aka eventual consistency) if we write the consumer code with that in mind then it should make things more robust and more scalable, rather than attempting to make sure everything progresses lockstep.